### ISSUE TITLE
Release 3.3.0-146 possibly affected by CVE-2020-36518

### ISSUE THREAD

**Issue by nlohiaTW** [gocd-contrib/github-oauth-authorization-plugin#203]
I have run Blackduck scan on the release 3.3.0-146 and it highlighted 1 security risk and 2 operational risks.

**Security Risk**:
1. CVE-2020-36518 affects jackson-databind up to version 2.13.2.1 

Will it be possible to work on a patch which includes jackson-databind 2.13.2.2 ?

**Operational Risks**:
1. The release also uses jar org.jetbrains.annotation version 13.0. The latest version for this is 23.0 Is there any specific reason to use the much older version.
2. The scans also shows that jackson-core 2.13.1 jar is running 1 version behind.

I request you to consider upgrading these two dependencies.

**Comment by chadlwilson** [comment]
For the second bit, these are transitive dependencies - upgrading ones transitive dependencies means taking on the risk of incompatibilities, and increased operational effort managing versions which can be a larger operational risk that using versions which are known and validated with the given direct dependency. Scans lack this kind of real world context.

```
runtimeClasspath - Runtime classpath of source set 'main'.
+--- com.google.code.gson:gson:2.9.0
+--- org.kohsuke:github-api:1.303
|    +--- org.apache.commons:commons-lang3:3.9
|    +--- com.fasterxml.jackson.core:jackson-databind:2.13.1
|    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.13.1
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.13.1
|    |    |         +--- com.fasterxml.jackson.core:jackson-annotations:2.13.1 (c)
|    |    |         +--- com.fasterxml.jackson.core:jackson-core:2.13.1 (c)
|    |    |         \--- com.fasterxml.jackson.core:jackson-databind:2.13.1 (c)
|    |    +--- com.fasterxml.jackson.core:jackson-core:2.13.1
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.13.1 (*)
|    |    \--- com.fasterxml.jackson:jackson-bom:2.13.1 (*)
|    \--- commons-io:commons-io:2.8.0
\--- com.squareup.okhttp3:okhttp:4.9.3
     +--- com.squareup.okio:okio:2.8.0
     |    +--- org.jetbrains.kotlin:kotlin-stdlib:1.4.0 -> 1.4.10
     |    |    +--- org.jetbrains.kotlin:kotlin-stdlib-common:1.4.10
     |    |    \--- org.jetbrains:annotations:13.0
     |    \--- org.jetbrains.kotlin:kotlin-stdlib-common:1.4.0 -> 1.4.10
     \--- org.jetbrains.kotlin:kotlin-stdlib:1.4.10 (*)
```     

In the specific case here, in my view the plugin is not really vulnerable to CVE-2020-36518 because it only deserializes JSON that is returned from github.com or someone's enterprise Github, not arbitrary locations and/or user input.

Rather than overriding transitive dependency versions to make some scan noise go away, it'd probably be better to wait for https://github.com/hub4j/github-api/pull/1417 to be released to incorporate the patched transitive dependency.

**Comment by nlohiaTW** [comment]
Thank you @chadlwilson. This is super helpful in understanding the impact of risks mentioned.

**Comment by chadlwilson** [comment]
Fixed in #205 and released in https://github.com/gocd-contrib/github-oauth-authorization-plugin/releases/tag/v3.3.0-172
